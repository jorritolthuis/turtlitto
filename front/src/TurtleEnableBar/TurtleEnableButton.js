import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

/**
 * A button that keeps track whether the turtle is enabled.
 * It is reflected to the user in terms in terms of button appearance.
 * Author: S.A. Tanja
 * Author: H.E. van der Laan
 * Author: T.T.P. Franken
 *
 * Props:
 *  - enabled: a boolean flag specifying whether is enabled or not
 *  - onTurtleEnableChange: function to call when the turtle enable button is pressed
 *  - id: identifier of a turtle
 *  - className: gives the classname for css
 */
const TurtleEnableButton = props => {
  const { enabled, id, className, onTurtleEnableChange } = props;

  return (
    <Button
      className={className}
      isActive={enabled}
      onClick={onTurtleEnableChange}
    >
      {id}
    </Button>
  );
};

TurtleEnableButton.propTypes = {
  enabled: PropTypes.bool.isRequired,
  onTurtleEnableChange: PropTypes.func.isRequired
};

const Button = styled.button`
  border-style: ${props =>
    props.isActive
      ? props.theme.buttonBorderStyleActive
      : props.theme.buttonBorderStyle};
  border-color: ${props =>
    props.isActive ? "none" : props.theme.buttonBorder};
  background-color: ${props =>
    props.isActive ? props.theme.buttonActive : props.theme.button};
  font-size: 4rem;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 2rem;
  min-width: 7rem;
  flex: 1;
`;
Button.displayName = "TurtleEnableButton";

export default TurtleEnableButton;
