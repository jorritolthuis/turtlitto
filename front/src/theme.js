// these colors are exposed by the styled-components ThemeProvider
// and used within the application
export default {
  success: "#00ff00",
  successLighter: "rgba(153, 255, 153, 0.25)",
  error: "#FF0000",
  errorLighter: "rgba(255,153,153, 0.4)",
  background: "linear-gradient(270deg, #196ebd, #01b0dd)",
  baseButton: "#FFFFFF",
  baseButtonDisabled: "rgba(216,216,216, 0.5)",
  info: "#bdbdbd",
  title: "#FFFFFF",
  secondary: "#d3d3d3",
  border: "#ccc",
  shadow: "rgba(0, 0, 0, 0.17)",
  turtleEnableBar: "#ededed",
  turtleEnableButton: "#ededed",
  turtleEnableButtonActive: "silver",
  bottomBarButton: "#ededed",
  bottomBarButtonActive: "silver",
  notificationSuccess: "green",
  notificationWarning: "#ffcc00",
  notificationError: "Tomato",
  dropBallButton: "#ededed",
  dropBallButtonActive: "silver",
  inOutButton: "#ededed",
  inOutButtonActive: "silver",
  refboxMagenta: "magenta",
  refboxCyan: "cyan",
  refboxMagentaActive: "magenta",
  refboxCyanActive: "cyan"
};
