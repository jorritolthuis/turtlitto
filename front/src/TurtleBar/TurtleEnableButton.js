import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { buttonColor } from "../theme.js";

/**
 *
 * Author: S.A. Tanja
 * Author: H.E. van der Laan
 * Author: T.T.P. Franken
 *
 *
 * A button that keeps track whether a turtle is enabled or not.
 * Is reflected to the user in terms in terms of button appearance.
 *
 * Props:
 * - enabled: whether this turtle is enabled or not
 * - onDisable: function to call when disabling this turtle
 * - onEnable: function to call when enabling this turtle
 *
 */
const TurtleEnableButton = props => {
  if (props.enabled) {
    return (
      <Button className={props.className} isActive onClick={props.onDisable}>
        {props.id}
      </Button>
    );
  } else {
    return (
      <Button className={props} onClick={props.onEnable}>
        {props.id}
      </Button>
    );
  }
};

TurtleEnableButton.propTypes = {
  enabled: PropTypes.bool.isRequired,
  onDisable: PropTypes.func.isRequired,
  onEnable: PropTypes.func.isRequired
};

const Button = styled.button`
  border-style: ${props => (props.isActive ? "inset" : "solid")};
  border-color: ${props => (props.isActive ? "none" : buttonColor)};
  background-color: ${props => (props.isActive ? "silver" : buttonColor)};
  width: 16%;
  min-width: 7.5rem;
  height: 10rem;
  font-size: 4rem;
  flex: 1;
  margin: 0.1rem;
  &:active {
    background: "silver";
  }
`;

export default TurtleEnableButton;
