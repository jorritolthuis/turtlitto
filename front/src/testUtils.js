import React from "react";
import { ThemeProvider } from "styled-components";
import theme from "./theme";
import { shallow, mount } from "enzyme";

export const shallowWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = tree => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  });
};
