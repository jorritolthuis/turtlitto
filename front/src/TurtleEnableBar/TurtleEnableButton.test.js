import React from "react";
import TurtleEnableButton from "./TurtleEnableButton";
import { shallowWithTheme } from "../testUtils";
import theme from "../theme";

//Test_items: TurtleEnableButton.js
//Input_spec: -
//Output_spec: -
//Envir_needs: snapshot (automatically made, found in the __snapshot__ folder).
describe("TurtleEnableButton", () => {
  describe("is active", () => {
    it("should match snapshot", () => {
      const wrapper = shallowWithTheme(
        <TurtleEnableButton enabled id={1} onTurtleEnableChange={() => {}} />
      ).dive();

      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toHaveStyleRule("background-color", theme.buttonActive);
      expect(wrapper).toHaveStyleRule("border-color", "none");
      expect(wrapper).toHaveStyleRule(
        "border-style",
        theme.buttonBorderStyleActive
      );
    });
  });

  describe("is inactive", () => {
    it("should match snapshot", () => {
      const wrapper = shallowWithTheme(
        <TurtleEnableButton
          id={1}
          enabled={false}
          onTurtleEnableChange={() => {}}
        />
      ).dive();

      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toHaveStyleRule("background-color", theme.button);
      expect(wrapper).toHaveStyleRule("border-color", theme.buttonBorder);
      expect(wrapper).toHaveStyleRule("border-style", theme.buttonBorderStyle);
    });
  });
});
