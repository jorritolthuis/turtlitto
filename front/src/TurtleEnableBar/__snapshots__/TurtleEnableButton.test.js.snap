// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurtleEnableButton should match snapshot 1`] = `
.c0 {
  border-style: inset;
  border-color: none;
  background-color: silver;
  font-size: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2rem;
  min-width: 7rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<button
  className="c0"
  onClick={[Function]}
>
  1
</button>
`;
